### PR TITLE
Add keras 1.0.7

### PR DESCRIPTION
**Pull request by ericdill**

### CODE REVIEW

**Comment by ericdill** [comment]
@fchollet I am adding keras to conda-forge for linux, mac and windows. This will allow users to easily install this with [conda] (http://conda.pydata.org/docs/) (conda install -c conda-forge keras). The maintenance burden here is light as all testing and releases are built and deployed automatically with CI. Changes to how it is being built can be controlled by changing the recipe which will be located at https://github.com/conda-forge/keras-feedstock shortly after this PR is finished and merged. If you would like to be added as a maintainer (to make new releases for instance), I can happily add you to the maintainers list. If not, that is totally fine too!

**Comment by ericdill** [comment]
@conda-forge/core  Here's a potentially nasty question... keras runs automated tests on linux, but not windows/os-x.  Should I attempt to run the tests for keras on os-x/windows in the conda-forge CI builds?  My rationale for even proposing the question is that is seems difficult to guarantee that conda-forge is producing a high quality cross-platform packages for keras on windows/os-x without running the automated tests.  On the other hand it also feels out of scope for conda-forge to become a place that runs the test suite for projects for all platforms.  I am leaning towards automated testing on windows/os-x being the responsibility of the library authors and not conda-forge, but I can attempt to run the test suite on appveyor/travis if you think it's a good idea.

**Comment by fchollet** [comment]
> If you would like to be added as a maintainer (to make new releases for instance), I can happily add you to the maintainers list. If not, that is totally fine too!

Sorry, I won't have time for it, so you would have to take care of tracking new releases yourself. We basically never add new dependencies, so there shouldn't be much to do.

**Comment by jakirkham** [comment]
It's pure Python right? Normally we haven't bothered running the full test suite if it is pure Python. Especially if the authors are using CI as we assume there code is in good working order.

**Comment by ericdill** [comment]
> Sorry, I won't have time for it, so you would have to take care of tracking new releases yourself. 

No need to apologize.  I certainly understand that you are very busy!

> We basically never add new dependencies

Good to know, thanks!

**Comment by ericdill** [comment]
@jakirkham  

> It's pure Python right?

Yes, but at first glance there are non-trivial wrappers around [theano] (https://github.com/fchollet/keras/blob/master/keras/backend/theano_backend.py) and [tensorflow] (https://github.com/fchollet/keras/blob/master/keras/backend/tensorflow_backend.py).  Not exercising the theano/tensorflow wrapper on os-x and the theano wrapper on windows is the crux of what I am concerned about.

**Comment by patricksnape** [comment]
But surely that is a responsibility of the Windows Theano recipe rather than Keras itself? As long as Theano is confirmed to work on Windows conda-forge we can be pretty sure that Keras - as just a wrapper - will also work?

Also, this definitely seems like something Keras should be doing itself before we try to do it!

**Comment by ericdill** [comment]
OK I am convinced. Thanks for the feedback @jakirkham and @patricksnape

**Comment by ericdill** [comment]
This should be good to go, thanks all

File: recipes/keras/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+{% set name = "keras" %}
+{% set sha256 = "b94158c9dd09d3af79acf0ea06ed0084d4645b0ea1b62168f10929260731b14e" %}
+{% set github_owner = "fchollet" %}
+
+package:
+    name: {{ name }}
+    version: {{ version }}
+
+source:
+    url: https://github.com/{{ github_owner }}/{{ name }}/archive/{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipes/keras/meta.yaml]: I feel if we go this way `name` is not guaranteed to be the repo name. So it should either have a different jinja variable or this extra templating (including owner, not version) should be dropped.

File: recipes/keras/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+about:
+    home: https://github.com/fchollet/keras
+    license: MIT
+    summary: Deep Learning for Python
+
+extra:
+    recipe-maintainers:
+        - ericdill
+        - mariusvniekerk
+
```
Comment (jakirkham) [inline recipes/keras/meta.yaml]: nit: please drop extra newline.

File: recipes/keras/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    fn: {{ name }}-{{ version }}.tar.gz
+    sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+
+requirements:
+    build:
+        - toolchain
```
Comment (jakirkham) [inline recipes/keras/meta.yaml]: Does this have compiled code? I didn't think it did, but maybe I'm missing something.

File: recipes/keras/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    run:
+        - h5py
+        - numpy
+        - setuptools
+        - scipy
+        - tensorflow  # [not win]
+        - python
+        - pyyaml
+        - six
+        - theano
```
Comment (jakirkham) [inline recipes/keras/meta.yaml]: nit: please add empty line between `requirements` and `test`.

File: recipes/keras/meta.yaml (hunk 3 above)
Comment (ericdill) [inline recipes/keras/meta.yaml]: Nope, you're right. You're not missing anything.

File: recipes/keras/meta.yaml (hunk 1 above)
Comment (ericdill) [inline recipes/keras/meta.yaml]: Was "trying this on for size" to see how I liked it.  Turns out I don't like it that much.  Dropping github_owner and name.

File: recipes/keras/meta.yaml (hunk 3 above)
Comment (jakirkham) [inline recipes/keras/meta.yaml]: Ok, then let's drop this please.

**Comment by jakirkham** [comment]
Minor things it seems. So once addressed LGTM.

**Comment by ericdill** [comment]
Thanks for the review @jakirkham . Made all the suggested changes

File: recipes/keras/meta.yaml (hunk 4 above)
Comment (jakirkham) [inline recipes/keras/meta.yaml]: Done.

**Comment by jakirkham** [comment]
Thanks @ericdill. Will merge when it passes.

AppVeyor didn't start on its own. So had to start it manually.

**Comment by jakirkham** [comment]
Thanks @ericdill.

**Comment by ericdill** [comment]
Thanks @jakirkham !

**Pull request merged**